src/popup/components/PayWebsiteForm.tsx

```diff
@@ -42,54 +34,72 @@ export const PayWebsiteForm = () => {
 
   const form = React.useRef<HTMLFormElement>(null);
   const [isSubmitting, setIsSubmitting] = React.useState(false);
-  const [buttonState, setButtonState] =
-    React.useState<keyof typeof BUTTON_STATE>('idle');
-  const isIdle = React.useMemo(() => buttonState === 'idle', [buttonState]);
+  const [msg, setMsg] = React.useState<null | {
+    type: 'success' | 'warn';
+    message: string;
+  }>(null);
 
   const onSubmit = async (ev: React.FormEvent<HTMLFormElement>) => {
     ev.preventDefault();
-    if (buttonState !== 'idle') return;
+    if (isSubmitting) return;
     setErrors({ amount: null, pay: null });
+    setMsg(null);
 
-    setButtonState('loading');
     setIsSubmitting(true);
 
     const response = await message.send('PAY_WEBSITE', { amount });
 
     if (!response.success) {
-      setButtonState('idle');
-      setErrors((prev) => ({ ...prev, pay: toErrorInfo(response.message) }));
+      setErrors((prev) => ({
+        ...prev,
+        pay: toErrorInfo(response.error || response.message),
+      }));
     } else {
-      setButtonState('success');
       setAmount('');
+      const { type, url, sentAmountFormatted } = response.payload;
+      const msg = t('pay_state_success', [sentAmountFormatted, url]);
```

Comment:
I feel like adding website URL again in the message is not really needed since the label explicitly says: `Pay {website}`.

Reply:
➕